# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerOp.h
# Commit: 379cd5044768
# Full Hash: 379cd5044768c61ee7b94cc383fa498433a30999
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P1 Make WorkerRunnable to be a base class for runnables on Worker thread and Worker's parent thread. r=dom-worker-reviewers,asuth
#   
#   This is the first step in splitting the parent thread runnable out of WorkerRunnable.
#   
#   To reuse the runnable dispatching codes in Worker, we still need a base class for runnable on the worker thread and the parent thread.
# ==============================================================================

diff -r 29cb10d35b5a -r 379cd5044768 dom/serviceworkers/ServiceWorkerOp.h
--- a/dom/serviceworkers/ServiceWorkerOp.h	Fri Apr 19 09:32:48 2024 +0000
+++ b/dom/serviceworkers/ServiceWorkerOp.h	Fri Apr 19 09:41:57 2024 +0000
@@ -62,7 +62,8 @@
   bool IsTerminationOp() const;
 
   // Override to provide a runnable that's not a `ServiceWorkerOpRunnable.`
-  virtual RefPtr<WorkerRunnable> GetRunnable(WorkerPrivate* aWorkerPrivate);
+  virtual RefPtr<WorkerThreadRunnable> GetRunnable(
+      WorkerPrivate* aWorkerPrivate);
 
   // Overridden by ServiceWorkerOp subclasses, it should return true when
   // the ServiceWorkerOp was executed successfully (and false if it did fail).